core/src/smartcontracts/wasm/mod.rs

```diff
@@ -284,153 +356,79 @@ impl<'wrld> Runtime<'wrld> {
         }
     }
 
-    fn query_authority(mut caller: Caller<State>) -> Result<WasmUsize, Trap> {
-        let memory = Self::get_memory(&mut caller)?;
-        let alloc_fn = Self::get_alloc_fn(&mut caller)?;
-        let state = caller.data();
-        let authority = &state.account_id;
-
-        let bytes = Self::encode_with_length_prefix(authority)?;
-        let authority_offset =
-            Self::encode_bytes_into_memory(&bytes, &memory, &alloc_fn, &mut caller)?;
-        Ok(authority_offset)
+    #[codec::wrap]
+    fn query_authority(state: &State) -> AccountId {
+        state.account_id.clone()
     }
 
-    fn query_triggering_event(mut caller: Caller<State>) -> Result<WasmUsize, Trap> {
-        let memory = Self::get_memory(&mut caller)?;
-        let alloc_fn = Self::get_alloc_fn(&mut caller)?;
-        let state = caller.data();
-        let event = state
-            .triggering_event
-            .as_ref()
-            .ok_or_else(|| Trap::new("There is no triggering event".to_owned()))?;
-
-        let bytes = Self::encode_with_length_prefix(event)?;
-        let event_offset = Self::encode_bytes_into_memory(&bytes, &memory, &alloc_fn, &mut caller)?;
-        Ok(event_offset)
+    #[codec::wrap]
+    fn query_triggering_event(state: &State) -> Option<Event> {
+        state.triggering_event.clone()
     }
 
-    fn query_operation_to_validate(mut caller: Caller<State>) -> Result<WasmUsize, Trap> {
-        let memory = Self::get_memory(&mut caller)?;
-        let alloc_fn = Self::get_alloc_fn(&mut caller)?;
-        let state = caller.data();
-        let operation = state
-            .operation_to_validate
-            .as_ref()
-            .ok_or_else(|| Trap::new("There is no operation to validate".to_owned()))?;
-
-        let bytes = Self::encode_with_length_prefix(operation)?;
-        let operation_offset =
-            Self::encode_bytes_into_memory(&bytes, &memory, &alloc_fn, &mut caller)?;
-        Ok(operation_offset)
+    #[codec::wrap]
+    fn query_operation_to_validate(state: &State) -> Option<NeedsValidationBox> {
+        state.operation_to_validate.clone()
     }
 
     fn query_max_log_level() -> u32 {
         iroha_logger::layer::max_log_level() as u32
     }
 
-    /// Log the given string at the given log level. When this function
-    /// is called, the module serializes the string to linear memory and
-    /// provides log level, offset and length as parameters
-    ///
-    /// # Warning
-    ///
-    /// This function doesn't take ownership of the provided
-    /// allocation
+    /// Log the given string at the given log level
     ///
     /// # Errors
     ///
     /// If log level or string decoding fails
-    fn log(
-        mut caller: Caller<State>,
-        log_level: u32,
-        offset: WasmUsize,
-        len: WasmUsize,
-    ) -> Result<(), Trap> {
+    #[codec::wrap]
+    fn log((log_level, msg): (u8, String), state: &State) -> Result<(), Trap> {
```

Comment:
do you think we can just use `panic!` or `assert!` and not return a `Trap`. I would prefer that because it indicates an unrecoverable error. This was a mistake on my part that I've implemented this to return `Result`

Reply:
```suggestion
    fn log((log_level, msg): (u8, String), state: &State) {
```

function signature should look like this

Reply:
Created #3555